packages/zod-form-data/README.md

```diff
@@ -163,6 +163,35 @@ the `FormData` will not include an entry for `myCheckbox` at all.
 
 ([Further reading](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#value))
 
+### file
+
+Transforms any empty File objects to `undefined` before validating.
+This makes it so empty files will fail required checks,
+allowing you to use `optional` for optional fields.
+If you call `zfd.file` with no arguments, it will assume the field is a required file by default.
+
+#### Usage
+
+```ts
+const schema = zfd.formData({
+  requiredFile: zfd.file(),
+  optional: zfd.file().optional(),
+})
+```
+
+There is a unique case in Remix when using a CustomUploadHandler, 
+the field will be a `File` on the client side, but an ID string (or URL) after uploading on the server.
+
+In this case you will need the schema to switch to string on the server:
+
+```ts
+const schema = (clientSide = true) => zfd.formData({
+  file: clientSide ? zfd.file() : zfd.file(z.string()),
+})
+```
+
+*Note: This will return `File | string` for the type. TODO: Example of type safety for this* 
```

Comment:
I think I figured this out. If we make `zfd.formData` accept a schema as well as an object, then you'll be able to compose the schemas better. This is what I wrote for an integration test for the file input (not using `zfd.formData` but same ida)

```ts
const baseSchema = z.object({
  description: zfd.text(),
});

const clientValidator = withZod(
  baseSchema.and(
    z.object({
      myFile: zfd.file(
        z.instanceof(File, {
          message: "Please choose a file",
        })
      ),
    })
  )
);

const serverValidator = withZod(
  baseSchema.and(
    z.object({
      myFile: zfd.file(z.string()),
    })
  )
);
```